### PR TITLE
Task completion prizes related on santa types values without any completion order

### PR DESCRIPTION
**Pull request by ice-myles**
Task completion prizes related on santa types values without any completion order

### CODE REVIEW

File: tokenomics/balance.go [hunk 1]
```diff
@@ -278,12 +279,12 @@ func (s *completedTasksSource) Process(ctx context.Context, message *messagebrok
 	if ctx.Err() != nil || len(message.Value) == 0 {
 		return errors.Wrap(ctx.Err(), "unexpected deadline while processing message")
 	}
-	const requiredCompletedTasks, adoptionMultiplicationFactor = 6, 168
```
Comment (ice-ares) [inline tokenomics/balance.go:281]: Dont u still need this? Cuz thats a prize value for completing the last remaining task

File: tokenomics/balance.go [hunk 2]
```diff
@@ -319,7 +320,10 @@ func (s *completedTasksSource) Process(ctx context.Context, message *messagebrok
 
 		return errors.Wrapf(err, "failed to get GetAdoptionSummary for id:%v", id)
 	}
-	prize := s.cfg.BaseMiningRate(time.Now(), res[0].CreatedAt) * adoptionMultiplicationFactor
+	if _, ok := tasks.TaskPrize[tasks.Type(val.Type)]; !ok {
```
Comment (ice-ares) [inline tokenomics/balance.go]: Do all tasks award coins?

Comment (ice-ares) [reply tokenomics/balance.go]: @ice-myles ?

File: tokenomics/balance.go [hunk 3]
```diff
@@ -278,12 +279,12 @@ func (s *completedTasksSource) Process(ctx context.Context, message *messagebrok
...
 	var val struct {
 		UserID         string `json:"userId" example:"edfd8c02-75e0-4687-9ac2-1ce4723865c4"`
+		Type           string `json:"type" example:"claim_username"`
 		CompletedTasks uint64 `json:"completedTasks,omitempty" example:"3"`
 	}
-	if err = json.UnmarshalContext(ctx, message.Value, &val); err != nil || val.UserID == "" || val.CompletedTasks != requiredCompletedTasks {
+	if err = json.UnmarshalContext(ctx, message.Value, &val); err != nil || val.UserID == "" {
 		return errors.Wrapf(err, "process: cannot unmarshall %v into %#v", string(message.Value), &val)
 	}
 	duplGuardKey := fmt.Sprintf("completed_tasks_ice_prize_dupl_guards:%v", val.UserID)
```
Comment (ice-ares) [inline tokenomics/balance.go]: add val.Type as well in the template

File: tokenomics/balance.go (hunk 1 above)
Comment (ice-ares) [reply tokenomics/balance.go:281]: @ice-myles ?

File: tokenomics/balance.go [hunk 4]
```diff
@@ -319,7 +320,10 @@ func (s *completedTasksSource) Process(ctx context.Context, message *messagebrok
 
 		return errors.Wrapf(err, "failed to get GetAdoptionSummary for id:%v", id)
 	}
-	prize := s.cfg.BaseMiningRate(time.Now(), res[0].CreatedAt) * adoptionMultiplicationFactor
```
Comment (ice-ares) [inline tokenomics/balance.go:322]: if u don't need createdAt, then remove the call to get it

Comment (ice-ares) [reply tokenomics/balance.go:322]: I also noticed that part is useless: 

if err != nil {
		return errors.Wrap(err, "failed to getCurrentAdoption")
	}

File: tokenomics/balance.go (hunk 1 above)
Comment (ice-myles) [reply tokenomics/balance.go:281]: It was said that we need to have the prize for each specific task without the order and 

> we don't need prize for doing all tasks

so do we need to become attached to last task? As I see no. we have 4 tasks with prizes 100, 200, 300, 400, user completed 2nd - got 200 amount, completed 4 - already 600, completed 1st - 700 and when he has completed the 3d he will just have + 300 = 1000 im summary.

**Pull request merged**